flytekit/core/node.py

```diff
@@ -84,7 +84,7 @@ def metadata(self) -> _workflow_model.NodeMetadata:
 
     def with_overrides(self, *args, **kwargs):
         if "node_name" in kwargs:
-            self._id = kwargs["node_name"]
```

Comment:
can you leave a comment here pointing to https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names as the spec for pod names? 

Also, worth noting that node names usually map to pod names, but not always. We don't have worry about this here though (i.e. it's fine to have `_dnsify` here).